python/pyspark/sql/readwriter.py

```diff
@@ -431,6 +432,8 @@ def jdbc(self, url, table, column=None, lowerBound=None, upperBound=None, numPar
         if column is not None:
             if numPartitions is None:
                 numPartitions = self._spark._sc.defaultParallelism
```

Comment:
This is contradicting with the scala version. Could you also change it to the following code
```Python
assert numPartitions is not None, "numPartitions can not be None when ``column`` is specified"
```